# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/mathieu/streamtuner/pythonplugin-segfault.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -ruN streamtuner-0.99.99/src/plugins/python/pst-main.c streamtuner-0.99.99-new/src/plugins/python/pst-main.c
--- streamtuner-0.99.99/src/plugins/python/pst-main.c	2004-12-11 03:00:39.000000000 +0100
+++ streamtuner-0.99.99-new/src/plugins/python/pst-main.c	2005-06-15 13:13:03.000000000 +0200
@@ -94,6 +94,7 @@
 {
   gboolean status = FALSE;
   PyObject *module;
+  char *argv[] = { "" };
 
   if (! check_api_version(err))
     return FALSE;
@@ -109,6 +110,7 @@
     }
 
   Py_Initialize();
+  PySys_SetArgv(G_N_ELEMENTS(argv), argv);
   PyEval_InitThreads();
 
   module = PyImport_AddModule("__main__");
